### ISSUE TITLE
Complete `Filecoin.EthGetTransactionReceipt` implementation

### ISSUE THREAD

**Issue by elmattic** [ChainSafe/forest#4759]
## Issue summary

This is a follow-up issue for #4691.

The RPC method still needs to be fully implemented and thoroughly tested. What is missing is the support of logs in `EthTxReceipt`.

## Task summary

- ~~[ ] Implement support for Events #4740~~
- [x] Re-introduce following methods for `EthEventHandler` *:
  - `eth_get_events_for_filter`
  - `wait_for_height_processed`
  - `parse_eth_filter_spec`
  - `get_eth_logs_for_block_and_transaction`
- [x] Re-introduce following functions **:
  - `eth_bloom_set`
  - `eth_filter_logs_from_events`
  - `eth_log_from_event`
  - `eth_tx_hash_from_message_cid`
  - `eth_tx_hash_from_signed_message`
- ~~[ ] Implement optional export of receipts in snapshots~~
- [x] Publicly expose `GasOutputs` in ref-fvm
- [x] Re-enable test by removing it from our filter lists

## Acceptance Criteria

The API compare test is passing and `EthGetTransactionReceipt` is removed from our filter lists.

## Other information and links

*: https://github.com/ChainSafe/forest/pull/4691/commits/c2281950a307a582357dc62108f880adc04603ca
**: https://github.com/ChainSafe/forest/pull/4691/commits/52a06dbe3b9ed14b88ab457358bc6f39d1efebfe

**Comment by LesnyRumcajs** [comment]
Blocked by #4740

**Comment by elmattic** [comment]
This issue is now unblocked, thanks to the spike.

**Comment by LesnyRumcajs** [comment]
Blocked by #4780

**Comment by elmattic** [comment]
This issue is now unblocked.

**Comment by elmattic** [comment]
Let’s skip exporting receipts and focus on removing duplicate code between Forest and the ref-fvm.

The next release should expose it (see https://github.com/filecoin-project/ref-fvm/pull/2091).

**Comment by LesnyRumcajs** [comment]
Nice! New release should be done soonish (NV25 feature flag removal)
